kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/NMDbusConnectorTest.java

```diff
@@ -0,0 +1,575 @@
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.lang.reflect.Field;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+import org.eclipse.kura.net.NetInterface;
+import org.eclipse.kura.net.NetInterfaceAddress;
+import org.eclipse.kura.net.NetInterfaceStatus;
+import org.eclipse.kura.net.NetInterfaceType;
+import org.eclipse.kura.net.NetworkService;
+import org.eclipse.kura.net.status.NetworkInterfaceStatus;
+import org.eclipse.kura.net.status.NetworkInterfaceType;
+import org.eclipse.kura.usb.UsbNetDevice;
+import org.freedesktop.NetworkManager;
+import org.freedesktop.dbus.DBusPath;
+import org.freedesktop.dbus.connections.impl.DBusConnection;
+import org.freedesktop.dbus.exceptions.DBusException;
+import org.freedesktop.dbus.interfaces.Properties;
+import org.freedesktop.dbus.types.UInt32;
+import org.freedesktop.dbus.types.Variant;
+import org.freedesktop.networkmanager.Device;
+import org.freedesktop.networkmanager.GetAppliedConnectionTuple;
+import org.freedesktop.networkmanager.Settings;
+import org.freedesktop.networkmanager.device.Generic;
+import org.freedesktop.networkmanager.settings.Connection;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+public class NMDbusConnectorTest {
+
+    DBusConnection dbusConnection = mock(DBusConnection.class, RETURNS_SMART_NULLS);
+    NetworkManager mockedNetworkManager = mock(NetworkManager.class);
+    NMDbusConnector instanceNMDbusConnector;
+    DBusConnection dbusConnectionInternal;
+
+    Boolean hasDBusExceptionBeenThrown = false;
+    Boolean hasNoSuchElementExceptionThrown = false;
+    Boolean hasNullPointerExceptionThrown = false;
+    
+    NetworkInterfaceStatus netInterface;
+    NetworkService networkService;
+    
+    Map<String, Device> mockDevices = new HashMap<>();
+
+    List<String> internalStringList;
+    Map<String, Object> netConfig = new HashMap<>();
+
+    @Before
+    public void setUpPersistantMocks() throws DBusException {
+        givenBasicMockedDbusConnector();
+    }
+    
+    @After
+    public void tearDown() {
+        resetSingleton(NMDbusConnector.class, "instance");
+    }
+
+    public static <T> void resetSingleton(Class<T> clazz, String fieldName) {
+        Field instance;
+        try {
+            instance = clazz.getDeclaredField(fieldName);
+            instance.setAccessible(true);
+            instance.set(null, null);
+            instance.setAccessible(false);
+        } catch (Exception e) {
+            throw new RuntimeException();
+        }
+    }
+
+    @Test
+    public void getDbusConnectionShouldWork() throws DBusException {
+        whenGetDbusConnectionIsRun();
+
```

Comment:
Remove the use of `@Before` such that we are more Gerkhin style compliant

 ```suggestion
    @After
    public void tearDown() {
        resetSingleton(NMDbusConnector.class, "instance");
    }

    public static <T> void resetSingleton(Class<T> clazz, String fieldName) {
        Field instance;
        try {
            instance = clazz.getDeclaredField(fieldName);
            instance.setAccessible(true);
            instance.set(null, null);
            instance.setAccessible(false);
        } catch (Exception e) {
            throw new RuntimeException();
        }
    }

    @Test
    public void getDbusConnectionShouldWork() throws DBusException {
        givenBasicMockedDbusConnector();
        whenGetDbusConnectionIsRun();

```